### PR TITLE
Add querier with new PromQL engine

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR adds a new `observatorium-volcano-query` deployment which deploys Thanos Querier with the new PromQL engine enabled. This also adds some new params to allow a different Thanos image to be used while deploying these new Queriers. This querier is not exposed to tenants.

I'm not mirroring traffic to this yet, as the new engine still doesn't support many functions, modifiers, or even functions within functions.
Mirroring traffic would lead to several "not implemented" errors right now, but can be enabled soon once some WIP stuff is merged upstream. For now, it would be valuable to process known queries and compare results with old Querier. 🙂

### CODE REVIEW

**Review by bwplotka — APPROVED** [review]
Amazing, thanks!

File: resources/services/observatorium-metrics-template.yaml [hunk 1]
```diff
@@ -990,6 +992,188 @@ objects:
...
+  metadata:
+    labels:
+      app.kubernetes.io/component: query-layer
+      app.kubernetes.io/instance: observatorium
+      app.kubernetes.io/name: thanos-query
+      app.kubernetes.io/part-of: observatorium
+      app.kubernetes.io/version: ${THANOS_VOLCANO_IMAGE_TAG}
+    name: observatorium-volcano-query
+  spec:
+    replicas: 2
```
Comment (matej-g) [inline resources/services/observatorium-metrics-template.yaml:1006]: We could parametrize this as well, but probably for playing around 2 replicas are ample :+1:

**Review by matej-g — APPROVED** [review]
Looks good! Even though I'm not a big fan of temporary changes, given this is an important feature to test, I'm happy to have it in. We could think in the future about adding something like a canary deployment, that we could scale / configure to test experimental features more easily.

File: resources/services/observatorium-metrics-template.yaml (hunk 1 above)
Comment (bwplotka) [reply resources/services/observatorium-metrics-template.yaml:1006]: 👍🏽

**Comment by fpetkovski** [comment]
@saswatamcode feel free to raise a separate PR with the improved fallback mechanism so that you can get unblocked on getting some real world traffic :)

**Pull request merged**